htdocs/cashdesk/class/Facturation.class.php

```diff
@@ -117,7 +117,12 @@ public function ajoutArticle()
         }
 
         // Define part of HT, VAT, TTC
-        $resultarray=calcul_price_total($this->qte, $this->prix(), $this->remisePercent(), $txtva, -1, -1, 0, 'HT', $use_npr, $product->type, $mysoc, $localtaxarray);
+
+		// Local Taxes
+		$localtax1_tx = get_localtax($txtva, 1, $societe);
```

Comment:
This looks strange to me.
First $localtax1_tx = get_localtax($txtva, 1, $societe);
should be 
$localtax1_tx = get_localtax($txtva, 1, $societe, $mysoc);

Then 
calcul_price_total() should use -1 as parameter for localtax1_tx to say use value already available into $localtaxarray
Usage of $localtaxarray is the future (it contains local tax rates but also the type + other information).
Is $localtaxarray not field correctly ?

Reply:
Ok, then it will be necessary to modify getLocalTaxesFromRate, since in Spain localtaxes do not depend on the seller, they depend on the buyer. I will make the changes

Reply:
It seems the method  getLocalTaxesFromRate  already tak into account the use of buyer for spain. I have this line with a comment question :

if ($mysoc->country_code == 'ES') $sql .= " WHERE t.fk_pays = c.rowid AND c.code = '".$buyer->country_code."'";    // local tax in spain use the buyer country ??